drivers/spi/Kconfig.rts5912

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2025, Realtek, SIBG-SD7
+# SPDX-License-Identifier: Apache-2.0
+
+config SPIM_REALTEK_RTS5912
```

Comment:
Rename `SPIM` to `SPI` and 'spim' to 'spi' throughout.

This also includes the compatible string: `realtek,rts5912-spi`